pkg/network/network_test.go

```diff
@@ -490,6 +490,121 @@ func TestConfiguration(t *testing.T) {
+	networkConfigTests := []struct {
+		name       string
+		wantErr    bool
+		wantConfig *Config
+		config     *corev1.ConfigMap
+		data       DomainTemplateValues
+	}{{
+		name:    "network configuration with annotations in template",
+		wantErr: false,
+		wantConfig: &Config{
+			IstioOutboundIPRanges:      "*",
+			DefaultClusterIngressClass: "foo-ingress",
+			DomainTemplate:             subdomainTemplate,
+			HTTPProtocol:               HTTPEnabled,
+		},
+		config: &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: system.Namespace(),
+				Name:      ConfigName,
+			},
+			Data: map[string]string{
+				IstioOutboundIPRangesKey:      "*",
+				DefaultClusterIngressClassKey: "foo-ingress",
+				DomainTemplateKey:             subdomainTemplate,
+			},
+		},
+		data: DomainTemplateValues{
+			Name:      "foo",
+			Namespace: "bar",
+			Annotations: map[string]string{
+				"sub": "sub1"},
+			Domain: "baz.com"},
+	}, {
+		name:    "network configuration using annotations in template with missing key",
+		wantErr: false,
+		wantConfig: &Config{
+			IstioOutboundIPRanges:      "*",
+			DefaultClusterIngressClass: "foo-ingress",
+			DomainTemplate:             subdomainTemplate,
+			HTTPProtocol:               HTTPEnabled,
+		},
+		config: &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: system.Namespace(),
+				Name:      ConfigName,
+			},
+			Data: map[string]string{
+				IstioOutboundIPRangesKey:      "*",
+				DefaultClusterIngressClassKey: "foo-ingress",
+				DomainTemplateKey:             subdomainTemplate,
+			},
+		},
+		data: DomainTemplateValues{
+			Name:      "foo",
+			Namespace: "bar",
+			Annotations: map[string]string{
+				"subfoo": "sub1"},
+			Domain: "baz.com"},
+	},
+		{
+			name:    "network configuration without annotations in template",
+			wantErr: false,
+			wantConfig: &Config{
+				IstioOutboundIPRanges:      "*",
+				DefaultClusterIngressClass: "foo-ingress",
+				DomainTemplate:             subdomainTemplate,
+				HTTPProtocol:               HTTPEnabled,
+			},
+			config: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: system.Namespace(),
+					Name:      ConfigName,
+				},
+				Data: map[string]string{
+					IstioOutboundIPRangesKey:      "*",
+					DefaultClusterIngressClassKey: "foo-ingress",
+					DomainTemplateKey:             subdomainTemplate,
+				},
+			},
+			data: DomainTemplateValues{
+				Name:      "foo",
+				Namespace: "bar",
+				Domain:    "baz.com"},
+		}}
+
+	for _, tt := range networkConfigTests {
+		t.Run(tt.name, func(t *testing.T) {
+			actualConfig, err := NewConfigFromConfigMap(tt.config)
+			if (err != nil) != tt.wantErr {
+				t.Fatalf("Test: %q; NewConfigFromConfigMap() error = %v, WantErr %v",
+					tt.name, err, tt.wantErr)
+			}
+			if tt.wantErr {
+				return
+			}
+
+			want := mustExecute(t, tt.wantConfig.GetDomainTemplate(), tt.data)
```

Comment:
As I commented above, I don't think the test should execute the template for both the "want" and "got", otherwise we're not actually testing anything since they're both going to evaluate to the same thing since all of the tests seem to use the same value in both spots. Put the hardcoded expected domain string in the "want" section so we can verify the templating worked as expected.